commons/marmotta-sesame-tools/marmotta-rio-rdfhdt/src/main/java/org/apache/marmotta/commons/sesame/rio/rdfhdt/RDFHDTParser.java

```diff
@@ -0,0 +1,183 @@
+	}
+
+	/**
+	 * Creates a new RDFParserBase that will use the supplied ValueFactory to
+	 * create RDF model objects.
+	 *
+	 * @param valueFactory
+	 *            A ValueFactory.
+	 */
+	public RDFHDTParser(ValueFactory valueFactory) {
+		super(valueFactory);
+	}
+
+	/**
+	 * Gets the RDF format that this parser can parse.
+	 */
+	@Override
+	public RDFFormat getRDFFormat() {
+		return RDFHDTFormat.FORMAT;
+	}
+
+	/**
+	 * Parses the data from the supplied InputStream, using the supplied baseURI
+	 * to resolve any relative URI references.
+	 *
+	 * @param in
+	 *            The InputStream from which to read the data.
+	 * @param baseURI
+	 *            The URI associated with the data in the InputStream.
+	 * @throws java.io.IOException
+	 *             If an I/O error occurred while data was read from the
+	 *             InputStream.
+	 * @throws org.openrdf.rio.RDFParseException
+	 *             If the parser has found an unrecoverable parse error.
+	 * @throws org.openrdf.rio.RDFHandlerException
+	 *             If the configured statement handler has encountered an
+	 *             unrecoverable error.
+	 */
+	@Override
+	public void parse(InputStream in, String baseURI) throws IOException,
+			RDFParseException, RDFHandlerException {
+		Preconditions.checkNotNull(baseURI);
+
+		setBaseURI(baseURI);
+
+		rdfHandler.startRDF();
+		DataInputStream input = new DataInputStream(new BufferedInputStream(in));
+		parseGlobalInfo(input);
+
+		rdfHandler.endRDF();
+	}
+
+	private void parseGlobalInfo(DataInputStream in) throws RDFParseException,
+			IOException {
+		parseMagic(in);
+		byte type = parseType(in);
+		byte globalInfoType = 1;
+		if (type != globalInfoType) {
+			reportFatalError("The global Information setion type should be: "
+					+ globalInfoType);
+		}
+		String format = parseFormat(in);
+		if(!RDFHDTConstants.HDT_CONTAINER.equals(format)){
+			throw new IllegalFormatException("This software cannot open this version of HDT File");
+		}
+		
+		parseProperties(in);
+	}
+
+	private void parseMagic(DataInputStream input) throws IOException,
+			RDFParseException {
+
+		// Check magic number
+		byte[] magicNumber = IOUtil.readBytes(input, MAGIC.length);
+		if (!Arrays.equals(magicNumber, MAGIC)) {
+			reportFatalError("File does not contain a binary RDF document");
+		}
+	}
+
+	private byte parseType(DataInputStream input) throws IOException {
+		return input.readByte();
+	}
+
+	private String parseFormat(DataInputStream input) throws IOException{
+		return parseString(input);
+
+	}
+	
+	private Properties parseProperties(DataInputStream input) throws IOException{
+		Properties properties = new Properties();
+        String propertiesStr = this.parseString(input);   
+        for(String item : propertiesStr.split(";")) {
+        	int pos = item.indexOf('=');
+        	if(pos!=-1) {
+        		String property = item.substring(0, pos);
+        		String value = item.substring(pos+1);
+        		properties.put(property, value);
```

Comment:
You get to the RDFHandler using the following check (it may be null):

    if(getRDFHandler() != null) {
        getRDFHandler().handleNamespace(prefix, uriString);
    }